pkg/broker/filter/filter_handler.go

```diff
@@ -206,34 +222,60 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 
 	h.reportArrivalTime(event, reportArgs)
 
-	h.send(ctx, writer, request.Header, subscriberURI.String(), reportArgs, event, ttl)
+	h.send(ctx, writer, request.Header, subscriberURI.URL(), reportArgs, event, ttl)
 }
 
-func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target string, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
+func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target *url.URL, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
 	// send the event to trigger's subscriber
-	response, err := h.sendEvent(ctx, headers, target, event, reportArgs)
-	if err != nil {
-		h.logger.Error("failed to send event", zap.Error(err))
-		writer.WriteHeader(http.StatusInternalServerError)
-		_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+	response, responseErr := h.sendEvent(ctx, headers, target, event, reportArgs)
+
+	if responseErr.err != nil {
+		h.logger.Error("failed to send event", zap.Error(responseErr.err))
+		// If error is not because of the response, it should respond with http.StatusInternalServerError
+		if responseErr.ResponseCode == NoResponse {
+
+			writer.WriteHeader(http.StatusInternalServerError)
+			_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+			return
+		}
+
+		writer.WriteHeader(responseErr.ResponseCode)
+		// Read Response body to responseErr
+		errExtensionInfo := broker.ErrExtensionInfo{
+			ErrDestination:  target,
+			ErrResponseBody: responseErr.ResponseBody,
+		}
+		errExtensionBytes, msErr := json.Marshal(errExtensionInfo)
```

Comment:
One problem with sending a JSON is that, Channel implementations that don't know about this will fail completely.

I wonder if we can respond with:

- body: ResponseBody
- an additional HTTP header the "knative error destination" with the target, and then channels can propagate the body  and the target by using those.

Reply:
Actually, this should be just fine.

Reply:
so, nevermind